test/e2e/ha/openstack_services_test.go

```diff
@@ -0,0 +1,226 @@
+package ha
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	contrail "github.com/Juniper/contrail-operator/pkg/apis/contrail/v1alpha1"
+	"github.com/Juniper/contrail-operator/pkg/controller/utils"
+	"github.com/Juniper/contrail-operator/test/logger"
+	"github.com/Juniper/contrail-operator/test/wait"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func TestHAOpenStackServices(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+	log := logger.New(t, "contrail", test.Global.Client)
+
+	if err := test.AddToFrameworkScheme(contrail.SchemeBuilder.AddToScheme, &contrail.ManagerList{}); err != nil {
+		t.Fatalf("Failed to add framework scheme: %v", err)
+	}
+
+	if err := ctx.InitializeClusterResources(&test.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval}); err != nil {
+		t.Fatalf("Failed to initialize cluster resources: %v", err)
+	}
+	namespace, err := ctx.GetNamespace()
+	assert.NoError(t, err)
+	f := test.Global
+
+	t.Run("given contrail operator is running", func(t *testing.T) {
+		err = e2eutil.WaitForOperatorDeployment(t, f.KubeClient, namespace, "contrail-operator", 1, retryInterval, waitForOperatorTimeout)
+		if err != nil {
+			log.DumpPods()
+		}
+		assert.NoError(t, err)
+
+		w := wait.Wait{
+			Namespace:     namespace,
+			Timeout:       waitTimeout,
+			RetryInterval: retryInterval,
+			KubeClient:    f.KubeClient,
+			Logger:        log,
+		}
+
+		cluster := getHAOpenStackCluster(namespace)
+
+		t.Run("when manager resource with Config and dependencies are created", func(t *testing.T) {
```

Comment:
I see that manager contains only Memcached right now :)

Reply:
changed description